addons/web/static/src/views/fields/properties/property_tags.js

```diff
@@ -0,0 +1,237 @@
+    }
+
+    /**
+     * Return the current tags that can be selected.
+     * Make a deep copy to not make change on the original object
+     * and to be able to discard change.
+     */
+    get currentTags() {
+        return JSON.parse(JSON.stringify(this.props.tags || []));
+    }
+
+    /**
+     * Options available in the autocomplete component.
+     */
+    get autocompleteActions() {
+        return [{
+            options: (request) => {
+                const tagsFiltered = this.props.tags.filter(tag =>
+                    (!this.props.value || this.props.value.indexOf(tag[0]) < 0)
+                    && (
+                        !request || !request.length
+                        || tag[1].toLocaleLowerCase().indexOf(request.toLocaleLowerCase()) >= 0
+                    )
+                );
+                if (!tagsFiltered || !tagsFiltered.length) {
+                    // no result, ask the user if he want to create a new tag
+                    if (!request || !request.length) {
+                        return [{
+                            value: null,
+                            label: _lt('Start typing...'),
+                            classList: 'fst-italic',
+                        }];
+                    } else if (!this.props.canChangeTags) {
+                        return [{
+                            value: null,
+                            label: _lt('No result'),
+                            classList: 'fst-italic',
+                        }];
+                    }
+
+                    return [{
+                        value: {toCreate: true, value: request},
+                        label: sprintf(_lt('Create "%s"'), request),
+                        classList: 'o_field_property_dropdown_add',
+                    }];
+                }
+                return tagsFiltered.map((tag) => {
+                    return {
+                        value: tag[0],
+                        label: tag[1],
+                    };
+                });
+            },
+        }];
+    }
+
+    /**
+     * Add one value in the current tag list values.
+     */
+    onOptionSelected(tagValue) {
+        if (!tagValue) {
+            // clicked on "Start typing..."
+            return;
+        }
+
+        if (tagValue.toCreate) {
+            this.onNewTag(tagValue.value);
+        } else {
+            const currentValue = this.currentValue;
+            const newValue = [...currentValue, tagValue];
+            this.props.onValueChange(newValue);
+        }
+    }
+
+    /**
+     * Ask to create a new tag that will be added in
+     * the definition and automatically selected.
+     */
+    onNewTag(newLabel) {
+        if (!newLabel || !newLabel.length) {
+            return;
+        }
+
+        const newValue = this.labelToValue(newLabel);
+
+        const existingTags = this.props.tags.find(tag => tag[0] === newValue);
+        if (existingTags) {
```

Comment:
Note that it may be confusing for the end user as "tag 1" and "tag_1" will have the same id but not the same name.

Not a big deal though.